### PR TITLE
Show category filter dropdown in public dashboards

### LINKED ISSUE #5060: Public dashboard does not have field values for a category filter
Using a public link for a dashboard with a category filter, the publicly viewed dashboard does not show the field values.

Inside MB
screen shot 2017-05-10 at 11 46 25 am

Public Link
screen shot 2017-05-10 at 11 46 38 am

### PR DESCRIPTION
**Pull request by attekei**
Resolves #5060.

Todo:
- [x] Write a test for `fetchDashboard` action
- [x] Don't call `fetchFieldValues` metadata action unnecessarily if we already have the field values

### CODE REVIEW

**Comment by salsakran** [comment]
How easy is this to test?

**Comment by attekei** [comment]
Easiest way: mock `PublicApi.dashboard` endpoint and provide a realistic response (just copy-paste from some real `/api/public/dashboard/:uuid` call). Check if parameter values are correctly added to the state tree using a relevant Redux selector.

Cooler but trickier way: If we're able to create a dashboard and make it programmatically (maybe by just calling the relevant redux actions for adding a card and a filter + toggling the sharing settings), we could test the actual UI interaction

**Comment by attekei** [comment]
Fixes also https://github.com/metabase/metabase/issues/5128 which is a duplicate of #5060

File: frontend/src/metabase/dashboard/dashboard.integ.spec.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import {
+    createTestStore,
+    login
+} from "metabase/__support__/integrated_tests";
+
+import { getParameterFieldValues } from "metabase/selectors/metadata";
+import { ADD_PARAM_VALUES } from "metabase/redux/metadata";
+
+// TODO Atte Keinänen 7/17/17: When we have a nice way to create dashboards in tests, this could use a real saved dashboard
+// instead of mocking the API endpoint
```
Comment (salsakran) [inline frontend/src/metabase/dashboard/dashboard.integ.spec.js]: We could probably just POST things to an API endpoint and create the dashboard in integration tests.

**Review by salsakran — APPROVED** [review]
Looks reasonable. Thanks for adding a test.

**Review by kdoh — APPROVED** [review]

**Pull request merged**